options/options.go

```diff
@@ -207,6 +209,15 @@ type TerragruntOptions struct {
 
 	// Controls if a module prefix will be prepended to TF outputs
 	IncludeModulePrefix bool
+
+	// Functions is an optional field which provides a way to add extra functions to the Terragrunt Evaluation Context.
+	// It specifies a function which will be called with the directory of the terragrunt.hcl which is being evaluated.
+	// It should return a map where keys are the names of Terragrunt functions, and values are the corresponding function
+	// implementations. These functions are incorporated into the hcl.EvalContext prior to evaluation.
+	// They always supersede the base functions from Terraform or Terragrunt. This means that the
+	// Functions field can be utilized to overwrite any base functions, which can be particularly
+	// beneficial for testing or cases where certain functions should not be executed.
+	Functions func(baseDir string) map[string]function.Function
```

Comment:
Not sure how much value brings passing only `baseDir` to function

I was thinking about creating a structure that will be passed as inputs with fields like:
 * filename
 * options.TerragruntOptions
 * baseDir

This in future will help to extend fields without affecting the function signature